src/pages/strategies/edit/index.tsx

```diff
@@ -1,25 +1,30 @@
 import { EditStrategyMain } from 'components/strategies/edit';
-import { PathNames, useNavigate } from 'libs/routing';
+import { PathNames, useNavigate, useParams } from 'libs/routing';
 import { useWeb3 } from 'libs/web3';
-import { useEffect } from 'react';
-import { useStore } from 'store';
+import { useEffect, useState } from 'react';
 import { StrategiesPage } from '..';
+import { Strategy, useGetUserStrategies } from 'libs/queries';
 
 export const EditStrategyPage = () => {
   const { user } = useWeb3();
-  const {
-    strategies: { strategyToEdit },
-  } = useStore();
+  const { data: strategies, isLoading } = useGetUserStrategies({ user });
+  const { strategyId } = useParams({ strict: false });
+  const [strategy, setStrategy] = useState<Strategy>();
   const navigate = useNavigate();
 
   useEffect(() => {
-    if (!user || !strategyToEdit) {
+    if (isLoading) return;
+    if (!user || !strategyId) {
       navigate({ to: PathNames.strategies });
+    } else {
+      const strategy = strategies?.find(({ id }) => id === strategyId);
+      if (strategy) setStrategy(strategy);
+      else navigate({ to: PathNames.strategies });
     }
-  }, [user, strategyToEdit, navigate]);
+  }, [user, strategyId, strategies, isLoading, navigate]);
 
-  return user && strategyToEdit ? (
-    <EditStrategyMain strategy={strategyToEdit} />
+  return user && strategy ? (
+    <EditStrategyMain strategy={strategy} />
```

Comment:
I think the loading state should be improved. Currently it shows the overview page. We should handle this a bit more elegant and show the loading animation.

And show a error message if the id is not found.

Reply:
I agree, but it would require new mockup for this state. This implementation match the current behavior (fallback to overview page), so I think we can address the loading/error state in another PR. What do you think @ashachaf ?